Source/WebCore/dom/EventDispatcher.cpp

```diff
@@ -172,6 +200,12 @@ void EventDispatcher::dispatchEvent(Node& node, Event& event)
         }
     }
 
+    if (hasNoEventListnerOrDefaultEventHandler) {
```

Comment:
This code path is taken when the target or the related target is inside a shadow tree such that we must compute EventPath to figure out whether the final target / related target will be still inside a shadow tree or not.